pkg/router/template/template_helper.go

```diff
@@ -386,6 +387,27 @@ func parseIPList(list string) string {
 	return list
 }
 
+// maxTimeoutFirstMatchedAndClipped finds the maximum timeout managed by a given annotation among all the routes, matches it against a given pattern, and clips.
+func maxTimeoutFirstMatchedAndClipped(aliases map[ServiceAliasConfigKey]ServiceAliasConfig, annotation, pattern string, values ...string) string {
+	var (
+		max         string
+		maxDuration time.Duration
+	)
+	for _, cfg := range aliases {
+		timeout := clipHAProxyTimeoutValue(firstMatch(pattern, append([]string{cfg.Annotations[annotation]}, values...)...))
+		if timeout != "" {
+			// No error handling because clipHAProxyTimeoutValue returns
+			// a valid timeout or an empty string. The latter is already handled.
+			timeoutDuration, _ := haproxytime.ParseDuration(timeout)
```

Comment:
haproxytime.ParseDuration is called from within clipHAProxyTimeoutValue.  Here you can use the `time` package's ParseDuration.
```suggestion
			timeoutDuration, _ := time.ParseDuration(timeout)
```

Reply:
Done.